lib/http.go

```diff
@@ -0,0 +1,50 @@
+package lib
+
+// Based on https://gist.github.com/josue/598ac048f78362f593f49436680b60b7
+
+import (
+	"log"
+	"net/http"
+	"strings"
+)
+
+/*
+GetURLHeaders returns a map array of all available headers.
+@param string - URL given
+@return map[string]interface{}
+*/
+func GetURLHeaders(url string) map[string]interface{} {
+	response, err := http.Head(url)
+	if err != nil {
+		log.Fatal("Error: Unable to download URL (", url, ") with error: ", err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		log.Fatal("Error: HTTP Status = ", response.Status)
+	}
+
+	headers := make(map[string]interface{})
```

Comment:
You can use a sizing hint for this map using make based on the length of response.Header.

```suggestion
	headers := make(map[string]interface{}, len(response.Header))
```